client/src/components/addSourceWizard/addSourceWizard.js

```diff
@@ -43,61 +126,534 @@ class AddSourceWizard extends React.Component {
       });
     };
 
-    Store.dispatch({
-      type: confirmationModalTypes.CONFIRMATION_MODAL_SHOW,
-      title: 'Cancel Add Source',
-      heading: 'Are you sure you want to exit this wizard?',
-      body: 'Exiting this wizard will cancel adding the source.',
-      cancelButtonText: 'No',
-      confirmButtonText: 'Yes',
-      onConfirm: onConfirm
+    stepsConfirmed().then(confirmed => {
+      if (confirmed) {
+        Store.dispatch({
+          type: sourcesTypes.UPDATE_SOURCE_HIDE
+        });
+      } else {
+        Store.dispatch({
+          type: confirmationModalTypes.CONFIRMATION_MODAL_SHOW,
+          title: 'Cancel Add Source',
+          heading: 'Are you sure you want to exit this wizard?',
+          body: 'Exiting this wizard will cancel adding the source.',
+          cancelButtonText: 'No',
+          confirmButtonText: 'Yes',
+          onConfirm: confirmationDialogAccepted
+        });
+      }
     });
   }
 
-  onStep(stepIndex) {
-    if (stepIndex === this.state.activeStepIndex) {
-      return;
-    }
-
-    this.setState({
-      activeStepIndex: stepIndex
-    });
-  }
+  onStep(stepIndex) {}
 
-  onNext() {
+  onNext(event, submit) {
     const numberSteps = addSourceWizardSteps.length;
     let { activeStepIndex } = this.state;
 
     if (activeStepIndex < numberSteps - 1) {
       this.setState({ activeStepIndex: (activeStepIndex += 1) });
     }
+
+    if (submit) {
+      console.log('SAVING FORM...');
+    }
   }
 
   onBack() {
     let { activeStepIndex } = this.state;
+    let calcStep;
 
     if (activeStepIndex >= 1) {
-      this.setState({ activeStepIndex: (activeStepIndex -= 1) });
+      calcStep = activeStepIndex - 1;
+
+      this.setState({ activeStepIndex: calcStep });
+
+      this.resetInitialState(null, calcStep);
     }
   }
 
-  onSubmit() {}
+  onSubmit(event) {
+    this.onNext(event, true);
+  }
 
-  onReset() {}
+  validateUpdateSourceType(sourceType) {
+    if (!sourceType) {
+      return 'You must select a source type';
+    }
+
+    return '';
+  }
+
+  updateSourceType(event) {
+    this.setState(
+      {
+        sourceType: event.target.value,
+        sourceTypeError: this.validateUpdateSourceType(event.target.value)
+      },
+      () => this.validateStepOne()
+    );
+  }
+
+  validateStepOne() {
+    const { sourceTypeError } = this.state;
+
+    if (sourceTypeError === '') {
+      this.setState({ stepOneValid: true });
+    }
+  }
+
+  renderStepOne() {
+    const { sourceType, sourceTypeError } = this.state;
+
+    return (
+      <Form horizontal>
+        <h3 className="right-aligned_basic-form">Select source type</h3>
+        <Form.FormGroup validationState={sourceTypeError ? 'error' : null}>
+          <Radio
+            name="sourceType"
+            value={'network'}
+            checked={sourceType === 'network'}
+            onChange={e => this.updateSourceType(e)}
```

Comment:
optional: might be better just to pass 'network' rather the the event. personal preference though.